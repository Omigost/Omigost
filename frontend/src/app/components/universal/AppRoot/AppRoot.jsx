import React from 'react'

import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import { connect, Provider } from 'react-redux'
import { Router, Route, browserHistory } from 'react-router'
import thunk from 'redux-thunk'
import { syncHistoryWithStore, routerReducer } from 'react-router-redux'
import reducers from 'reducers/reducers.js'
import * as actions from 'reducers/actions.js'
import routes from 'routes/routes.js'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
    combineReducers(
        Object.assign({}, reducers, {
            routing: routerReducer
        })
    ),
    composeEnhancers(applyMiddleware(thunk))
);

const history = syncHistoryWithStore(browserHistory, store);

class AppRouter extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inited: false
        };
    }

    componentDidMount() {
        this.setState({
            inited: true
        });
    }

    render() {
      return (
          <Router history={history}>
              {
                  routes
              }
          </Router>
      );
    }
}

const mapStateToProps = state => {
    return {
        content: state.app.content
    }
}

const mapDispatchToProps = dispatch => {
    window.loadPrefetchedContent = (content) => {
        console.warn("Install prefetched content");
        dispatch(actions.fetchedContent(Object.assign({}, content, {
            loaded: true
        })));
    };
    return {
        fetchedContent: (content) => dispatch(actions.fetchedContent(content))
    }
}

const AppRouterConnected = connect(
    mapStateToProps,
    mapDispatchToProps
)(AppRouter)


class AppRoot extends React.Component {
    render() {
        return (
            <Provider store={store}>
                <AppRouterConnected />
            </Provider>
        );
    }
}

export default AppRoot;