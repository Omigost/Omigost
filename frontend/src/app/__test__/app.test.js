import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

import AppRoot from 'components/AppRoot/AppRoot.jsx'

test('Snapshot test of App main component', () => {
    const component = mount(
        <AppRoot />
    );

    expect(toJson(component, {
        mode: 'deep'
    })).toMatchSnapshot();
});