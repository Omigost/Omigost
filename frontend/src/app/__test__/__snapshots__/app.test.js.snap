// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of App main component 1`] = `
<div
  className=""
  styleName="Page"
>
  <div
    styleName="Loading"
  >
    <img
      src="img/web_logo_eye_loading.svg"
      styleName="EyeGraph"
    />
  </div>
</div>
`;
